File: src/ZCalc.Blazor/Models/Molecule.cs
using ZCalc.Bonds;

namespace ZCalc.Blazor.Models;

public record Molecule
{
    public Atom[] Atoms { get; set; } = Array.Empty<Atom>();
    
    public Bond[] Bonds { get; set; }= Array.Empty<Bond>();
}

public record Atom {
    public string Element { get; set; } = String.Empty;
    
    public Point Point { get; set; } 
    
    public Color Color { get; set; }
}

File: src/ZCalc.Blazor/js/src/molecule.ts
export interface Molecule {
    atoms: Atom[]
    
    bonds: Bond[]
}

export interface Atom {
    element: string
    
    point: Point
    
    color: Color
}

export interface Point {
    x: number

    y: number

    z: number
}

export interface Color {
    r: number
    
    g: number
    
    b: number
}

export interface Bond {
    atom1: number

    atom2: number
}